File: Samples for Parallel Programming with the .NET Framework/[C#][VB][C++][F#]-Samples for Parallel Programming with the .NET Framework/C# and C++ and F# and VB/C#,C++,F#,VB/Raytracer/Raytracer_VisualBasic/Ray.vb
'--------------------------------------------------------------------------
' 
'  Copyright (c) Microsoft Corporation.  All rights reserved. 
' 
'  File: Ray.vb
'
'--------------------------------------------------------------------------

Public Structure Ray

    Public Start As Vector
    Public Direction As Vector

    Public Sub New(ByVal start As Vector, ByVal dir As Vector)
        Me.Start = start
        Me.Direction = dir
    End Sub

End Structure

File: Samples for Parallel Programming with the .NET Framework/[C#][VB][C++][F#]-Samples for Parallel Programming with the .NET Framework/C# and C++ and F# and VB/C#,C++,F#,VB/Raytracer/Raytracer_CSharp/Ray.cs
//--------------------------------------------------------------------------
// 
//  Copyright (c) Microsoft Corporation.  All rights reserved. 
// 
//  File: Ray.cs
//
//--------------------------------------------------------------------------

namespace Microsoft.ParallelComputingPlatform.ParallelExtensions.Samples
{
    struct Ray
    {
        public Vector Start;
        public Vector Dir;

        public Ray(Vector start, Vector dir) { Start = start; Dir = dir; }
    }
}
